### DISCUSSION TITLE
Differentiable Group Normalization.

### DISCUSSION THREAD

**Discussion by wsad1** [pyg-team/pytorch_geometric#2113] [Ideas]
## 🚀 Feature
This [paper](https://papers.nips.cc/paper/2020/hash/33dd6dba1d56e826aac1cbf23cdcca87-Abstract.html) proposes a new normalization technique. Where the nodes in a group are normalized independently and the group assignment is learnable. 

It also proposes two metrics to measure over smoothing. 
## Motivation
The results look good in comparison to no norm(NN), pair-norm(PN) and batch norm(BN). But the paper doesn't compare results of other normalizations techniques.
![image](https://user-images.githubusercontent.com/13963626/107850902-fbacff80-6e2b-11eb-99c2-503a56b969f7.png)

Further the metrics for over smoothing might be useful to analyze gcn's. 

## Additional Context
torch-geometric has a few normalization layers, so it might not be too useful to add this. But i feel the over smoothing metrics could be a good addition. Do let me know your thoughts on this.

**Comment by rusty1s** [comment]
Interesting, are you interested in contributing?

**Reply by wsad1** [reply]
Yes, i was planning to contribute this feature.
